### PR TITLE
chore: Replace deprecated DynamicFontLabel init with the new one using the new Wire Font Style - WPB-11467

### PR DESCRIPTION
**Pull request by agisilaos**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-11467" title="WPB-11467" target="_blank">TaskWPB-11467</a>  [iOS] Replace deprecated DynamicFontLabel init with the new one using the new Wire Font Style
  </td></table>
  <br />
 

With this PR we replace old deprecated init of `DynamicFontLabel` with new one using the new Wire Font Style. Many snapshot reference images has been updated to reflect those changes. A couple of more PRs will follow after this one to complete the migration. Doing it bit by bit so it's easier to review. 

Figma: [LInk](https://www.figma.com/design/exxrpYZ28N1WOylyw9TYh1/Accessibility-Quick-Wins-(iOS)?node-id=0-1&node-type=canvas&t=eWbEdqDa8LxnRDcV-0)

---

### Checklist

- [x] Title contains a reference JIRA issue number like `[WPB-XXX]`.
- [x] Description is filled and free of optional paragraphs.
- [x] Adds/updates automated tests.

---

### CODE REVIEW

File: wire-ios/Wire-iOS/Sources/UserInterface/SelfProfile/ProfileHeaderViewController.swift [hunk 1]
```diff
@@ -54,12 +54,12 @@ final class ProfileHeaderViewController: UIViewController {
...
     private let nameLabel: DynamicFontLabel = {
-        let label = DynamicFontLabel(fontSpec: .accountName, color: LabelColors.textDefault)
+        let label = DynamicFontLabel(style: .h2, color: LabelColors.textDefault)
         label.accessibilityLabel = AccountPageStrings.Name.description
         label.accessibilityIdentifier = "name"
 
-        label.setContentHuggingPriority(UILayoutPriority.required, for: .vertical)
-        label.setContentCompressionResistancePriority(UILayoutPriority.required, for: .vertical)
+//        label.setContentHuggingPriority(UILayoutPriority.required, for: .vertical)
+//        label.setContentCompressionResistancePriority(UILayoutPriority.required, for: .vertical)
```
Comment (caldrian) [inline wire-ios/Wire-iOS/Sources/UserInterface/SelfProfile/ProfileHeaderViewController.swift]: May the commented code be deleted?

File: wire-ios/Wire-iOS/Sources/UserInterface/SelfProfile/WarningLabelView.swift [hunk 2]
```diff
@@ -24,7 +24,7 @@ import WireDesign
 final class WarningLabelView: UIView {
     private let stackView = UIStackView(axis: .horizontal)
     private let imageView = UIImageView(image: UIImage(named: "Info"))
-    private let label = DynamicFontLabel(fontSpec: .mediumSemiboldFont,
+    private let label = DynamicFontLabel(style: .h5,
                                  color: SemanticColors.Label.textErrorDefault)
```
Comment (caldrian) [inline wire-ios/Wire-iOS/Sources/UserInterface/SelfProfile/WarningLabelView.swift]: not properly formatted

**Review by caldrian — APPROVED** [review]

File: wire-ios/Wire-iOS Tests/ReferenceImages/ConversationListHeaderViewSnapshotTests/testForBadgeNumberEquals10.1.png
Comment (samwyndham) [inline wire-ios/Wire-iOS Tests/ReferenceImages/ConversationListHeaderViewSnapshotTests/testForBadgeNumberEquals10.1.png:1]: @agisilaos In this test and others from the same test case there is quite a big font change. Is this intended?

**Review by samwyndham — APPROVED** [review]
Nice work!

File: wire-ios/Wire-iOS/Sources/UserInterface/SelfProfile/ProfileHeaderViewController.swift (hunk 1 above)
Comment (agisilaos) [reply wire-ios/Wire-iOS/Sources/UserInterface/SelfProfile/ProfileHeaderViewController.swift]: Fixed!

File: wire-ios/Wire-iOS/Sources/UserInterface/SelfProfile/WarningLabelView.swift (hunk 2 above)
Comment (agisilaos) [reply wire-ios/Wire-iOS/Sources/UserInterface/SelfProfile/WarningLabelView.swift]: Fixed!

File: wire-ios/Wire-iOS Tests/ReferenceImages/ConversationListHeaderViewSnapshotTests/testForBadgeNumberEquals10.1.png
Comment (agisilaos) [reply wire-ios/Wire-iOS Tests/ReferenceImages/ConversationListHeaderViewSnapshotTests/testForBadgeNumberEquals10.1.png:1]: Yep it's intended!

**Pull request merged**